src/lib/PageTracker.js

```diff
@@ -0,0 +1,410 @@
+
+        this.container.addEventListener('click', this.throttledActiveUserHandler);
+        this.container.addEventListener('focusin', this.throttledActiveUserHandler);
+        this.container.addEventListener('focusout', this.inactiveUserHandler);
+        this.container.addEventListener('keydown', this.throttledActiveUserHandler);
+        this.container.addEventListener('mouseenter', this.throttledActiveUserHandler);
+        this.container.addEventListener('mouseleave', this.inactiveUserHandler);
+        this.container.addEventListener('mousemove', this.throttledActiveUserHandler);
+        this.container.addEventListener('touchmove', this.throttledActiveUserHandler);
+        this.container.addEventListener('touchstart', this.throttledActiveUserHandler);
+        this.container.addEventListener('wheel', this.throttledActiveUserHandler);
+
+        this.userEventsBounded = true;
+    }
+
+    /**
+     * Unbinds DOM listeners from the container that manage if the user goes active or inactive.
+     *
+     * @private
+     * @return {void}
+     */
+    unbindIdleUserEvents() {
+        if (!this.container) return;
+
+        this.container.removeEventListener('click', this.throttledActiveUserHandler);
+        this.container.removeEventListener('focusin', this.throttledActiveUserHandler);
+        this.container.removeEventListener('focusout', this.inactiveUserHandler);
+        this.container.removeEventListener('keydown', this.throttledActiveUserHandler);
+        this.container.removeEventListener('mouseenter', this.throttledActiveUserHandler);
+        this.container.removeEventListener('mouseleave', this.inactiveUserHandler);
+        this.container.removeEventListener('mousemove', this.throttledActiveUserHandler);
+        this.container.removeEventListener('touchmove', this.throttledActiveUserHandler);
+        this.container.removeEventListener('touchstart', this.throttledActiveUserHandler);
+        this.container.removeEventListener('wheel', this.throttledActiveUserHandler);
+
+        this.userEventsBounded = false;
+    }
+
+    /**
+     *  Construct and emit the report advanced content insight event
+     *
+     * @private
+     * @return {void}
+     */
+    report() {
+        const pageViewUTCTimestamp = Date.now();
+        const timeToBeReported = pageViewUTCTimestamp - this.startTime;
+
+        // Report only time with a minimum threshold for this moment. This is to avoid a situation that the
+        // user is just spamming the scroll event to change pages and not send a lot of requests.
+        // Ideally, we can just batch that data and send it later...
+        if (timeToBeReported > REPORT_TRESHOLD) {
+            const { id: fileId, file_version: { id: fileVersionId } = null } = this.file;
+            // Add the current event to the events Array.
+            this.events.push({
+                userId: Number(this.userId),
+                timestamp: pageViewUTCTimestamp,
```

Comment:
Did we resolve the question around what timecode format we wanted to use?

Reply:
Is not final, but DP is still using timestamps